File: cases/taint/javascript/model-propagator-position-negative/Opaque.js
function dfb_source() { // DFB-SOURCE: model-propagator-position-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: model-propagator-position-sink

const _impl = {
  identity: function identity(first, second) {
    return second;
  }
};

const Opaque = {
  select: function select(first, second) {
    const target = "identity";
    return Reflect.get(_impl, target).apply(null, [first, second]);
  }
};

function run() {
  dfb_sink(Opaque.select(dfb_source(), "clean"));
}

File: cases/taint/python/model-propagator-position-positive/opaque.py
class _Impl:
    @staticmethod
    def second(first, second):
        return second


_impl = _Impl()


def select(first, second):
    name = "second"
    return getattr(_impl, name)(first, second)


def dfb_source():  # DFB-SOURCE: model-propagator-position-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: model-propagator-position-sink
    pass


def run():
    dfb_sink(select("clean", dfb_source()))
